components/onboarding_tasks/onboarding_tasks_manager.tsx

```diff
@@ -39,65 +41,100 @@ import {
+import security from 'images/security.svg';
+import sunglasses from 'images/sunglasses.svg';
+import wrench from 'images/wrench.svg';
+
+import {t} from '../../utils/i18n';
+
 import {OnboardingTaskCategory, OnboardingTaskList, OnboardingTasksName, TaskNameMapToSteps} from './constants';
+import {generateTelemetryTag} from './utils';
 
 const getCategory = makeGetCategory();
 
-const taskLabels = {
-    [OnboardingTasksName.CHANNELS_TOUR]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_learn_more_about_messaging'
-            defaultMessage='🔀 Learn about messaging'
-        />
-    ),
-    [OnboardingTasksName.BOARDS_TOUR]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.plan_sprint_with_kanban_style_boards'
-            defaultMessage='🎯 Plan a sprint with Kanban-style boards'
-        />),
-    [OnboardingTasksName.PLAYBOOKS_TOUR]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_resolve_incidents_faster_with_playbooks'
-            defaultMessage='🐛 Resolve incidents faster with playbooks'
-        />
-    ),
-    [OnboardingTasksName.INVITE_PEOPLE]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_invite_team_members'
-            defaultMessage='👋 Invite team members to the workspace'
-        />
-    ),
-    [OnboardingTasksName.COMPLETE_YOUR_PROFILE]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_complete_your_profile'
-            defaultMessage='📷 Complete your profile'
-        />
-    ),
-    [OnboardingTasksName.EXPLORE_OTHER_TOOLS]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.explore_other_tools_in_platform'
-            defaultMessage='⛰️ Explore other tools in the platform'
-        />
-    ),
-    [OnboardingTasksName.DOWNLOAD_APP]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_download_mm_apps'
-            defaultMessage='📱 Download the Desktop and Mobile Apps'
-        />
-    ),
-    [OnboardingTasksName.VISIT_SYSTEM_CONSOLE]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_visit_system_console'
-            defaultMessage='🥰️ Visit the System Console to configure your workspace'
-        />
-    ),
-    [OnboardingTasksName.START_TRIAL]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_start_enterprise_trial'
-            defaultMessage='🏢 Learn more about Enterprise-level high-security features'
-        />
-    ),
+const TaskListImage = styled.img`
+    width: 24px;
+    height: 24px;
+`;
+
+// eslint-disable-next-line react-hooks/rules-of-hooks
+
+interface OnBoardingTaskListTaskProps {
+    id: string;
+    svg: string;
+    message: string;
+}
+
+const getTaskDetails = {
+    [OnboardingTasksName.CHANNELS_TOUR]: {
+        id: 'onboardingTask.checklist.task_learn_more_about_messaging',
+        svg: channels,
+        message: 'Take a tour of Channels.',
```

Comment:
These messages should use the localized ones.

Reply:
I don't think there's any need for the `id` to be the exact same from the localized message, it should be an identifier for the `picture` element (key).
Also with this implementation the i18n scripts won't be catching the messages and we'll be prone to mistakes in translation. 

IMHO it could be just:
```
    id: 'task_learn_more_about_messaging',
    svg: channels,
    message: formatMessage({
        id: onboardingTask.checklist.task_learn_more_about_messaging,
        defaultMessage: 'Take a tour of Channels.',
    })
```